--- neutrino-hd/src/driver/framebuffer.h	2012-07-30 12:15:20.943710648 +0200
+++ neutrino-hd/src/driver/framebuffer.h	2012-09-25 21:19:28.041555256 +0200
@@ -140,7 +140,7 @@
 #ifdef FB_DEVICE
 		void init(const char * const fbDevice = FB_DEVICE);
 #else
-		void init(const char * const fbDevice = "/dev/fb/0");
+		void init(const char * const fbDevice = "/dev/fb0");
 #endif
 		int setMode(unsigned int xRes, unsigned int yRes, unsigned int bpp);
