--- neutrino-mp/src/driver/framebuffer_ng.h	2013-01-17 12:35:42.898537648 +0100
+++ neutrino-mp/src/driver/framebuffer_ng.h	2013-01-17 15:24:33.856774361 +0100
@@ -165,8 +165,11 @@
 #ifdef USE_NEVIS_GXA
 		void setupGXA(void);
 #endif
-
+#ifdef EVOLUX
+		void init(const char * const fbDevice = "/dev/fb0");
+#else
 		void init(const char * const fbDevice = "/dev/fb/0");
+#endif
 		int setMode(unsigned int xRes, unsigned int yRes, unsigned int bpp);
 
 
--- neutrino-mp/lib/libtuxtxt/tuxtxt.cpp	2013-01-17 15:20:11.259472213 +0100
+++ neutrino-mp/lib/libtuxtxt/tuxtxt.cpp	2013-01-17 15:24:17.904695263 +0100
@@ -1657,12 +1657,20 @@
 
 	fb = -1;
 #ifdef USE_FBPAN
+#ifdef EVOLUX
+        if ((fb=open("/dev/fb0", O_RDWR)) == -1)
+        {
+                perror("TuxTxt <open /dev/fb0>");
+                return 0;
+        }
+#else
         if ((fb=open("/dev/fb/0", O_RDWR)) == -1)
         {
                 perror("TuxTxt <open /dev/fb/0>");
                 return 0;
         }
 #endif
+#endif
 
 	rc = _rc;
 	CFrameBuffer *fbp = CFrameBuffer::getInstance();

--- neutrino-mp/src/nhttpd/tuxboxapi/dbox/controlapi.cpp	2013-01-17 15:37:50.036722406 +0100
+++ neutrino-mp/src/nhttpd/tuxboxapi/dbox/controlapi.cpp	2013-01-17 15:41:12.301725387 +0100
@@ -28,8 +28,11 @@
 #include "neutrinoapi.h"
 #include "controlapi.h"
 #include "lcdapi.h"
-
+#ifdef EVOLUX
+#define EVENTDEV "/dev/input/event1"
+#else
 #define EVENTDEV "/dev/input/event0"
+#endif
 //-----------------------------------------------------------------------------
 enum {	// not defined in input.h but used like that, at least in 2.4.22
 	KEY_RELEASED = 0,
@@ -806,7 +809,11 @@
   int evd = open(EVENTDEV, O_RDWR);
   if (evd < 0) {
     hh->SendError();
+#ifdef EVOLUX
+    perror("opening event1 failed");
+#else
     perror("opening event0 failed");
+#endif
     return;
   }
   if (rc_send(evd, sendcode, KEY_PRESSED) < 0){

--- neutrino-mp/src/driver/rcinput.cpp	2013-01-17 15:37:49.996722181 +0100
+++ neutrino-mp/src/driver/rcinput.cpp	2013-01-17 15:49:05.460071651 +0100
@@ -63,7 +63,11 @@
 #if HAVE_SPARK_HARDWARE
 /* this relies on event0 being the AOTOM frontpanel driver device
  * TODO: what if another input device is present? */
+#ifdef EVOLUX
+const char * const RC_EVENT_DEVICE[NUMBER_OF_EVENT_DEVICES] = {"/dev/input/nevis_ir", "/dev/input/event1"};
+#else
 const char * const RC_EVENT_DEVICE[NUMBER_OF_EVENT_DEVICES] = {"/dev/input/nevis_ir", "/dev/input/event0"};
+#endif
 #else
 //const char * const RC_EVENT_DEVICE[NUMBER_OF_EVENT_DEVICES] = {"/dev/input/nevis_ir", "/dev/input/event0"};
 const char * const RC_EVENT_DEVICE[NUMBER_OF_EVENT_DEVICES] = {"/dev/input/nevis_ir"};
