--- neutrino-hd/lib/libtuxtxt/tuxtxt.cpp	2012-09-17 21:23:35.581705237 +0200
+++ neutrino-hd/lib/libtuxtxt/tuxtxt.cpp	2012-09-25 21:27:08.764985990 +0200
@@ -1735,7 +1735,7 @@
 #ifdef FB_DEVICE
                 perror("TuxTxt <open " FB_DEVICE ">");
 #else
-                perror("TuxTxt <open /dev/fb/0>");
+                perror("TuxTxt <open /dev/fb0>");
 #endif
                 return 0;
         }
