--- InfoBarGenerics.py	2009-09-19 14:45:24.000000000 +0200
+++ InfoBarGenerics.py.new	2009-09-28 12:49:59.282354436 +0200
@@ -290,6 +290,8 @@
 				"zapDown": (self.zapDown, _("next channel")),
 				"historyBack": (self.historyBack, _("previous channel in history")),
 				"historyNext": (self.historyNext, _("next channel in history")),
+				"prevBouquet": (self.prevBouquet, _("previous bouquet")),
+				"nextBouquet": (self.nextBouquet, _("next bouquet")),
 				"openServiceList": (self.openServiceList, _("open servicelist")),
 			})
 
@@ -328,6 +330,16 @@
 	def openServiceList(self):
 		self.session.execDialog(self.servicelist)
 
+	def prevBouquet(self):
+		if self.servicelist.inBouquet():
+			self.servicelist.prevBouquet()
+			self.servicelist.zap()
+
+	def nextBouquet(self):
+		if self.servicelist.inBouquet():
+			self.servicelist.nextBouquet()
+			self.servicelist.zap()
+
 	def zapUp(self):
 		if self.servicelist.inBouquet():
 			prev = self.servicelist.getCurrentSelection()
@@ -1275,7 +1287,17 @@
 		self["InstantExtensionsActions"] = HelpableActionMap(self, "InfobarExtensions",
 			{
 				"extensions": (self.showExtensionSelection, _("view extensions...")),
-			}, 1) # lower priority
+# ------------------ Extra-Menu Patch Red----------------
+				"evopanel": (self.callEVOPanelRed, _("view EVOPanel...")),
+			}, 1)
+
+	def callEVOPanelRed(self):
+                from Plugins.Extensions.EVOPanel.plugin import EVOPanelButton, EVOPanelServiceList, EVOPanel
+	        EVOPanelButton(self,"red")
+	        EVOPanelServiceList(self,self.servicelist)
+		self.session.open(EVOPanel)
+                EVOPanelButton(self,"")
+# ------------------ Extra-Menu Patch Red----------------
 
 	def addExtension(self, extension, key = None, type = EXTENSION_SINGLE):
 		self.list.append((type, extension, key))
@@ -2568,7 +2590,6 @@
 			eDVBServicePMTHandler.eventNoResources: _("No free tuner!"),
 			eDVBServicePMTHandler.eventTuneFailed: _("Tune failed!"),
 			eDVBServicePMTHandler.eventNoPAT: _("No data on transponder!\n(Timeout reading PAT)"),
-			eDVBServicePMTHandler.eventNoPATEntry: _("Service not found!\n(SID not found in PAT)"),
 			eDVBServicePMTHandler.eventNoPMT: _("Service invalid!\n(Timeout reading PMT)"),
 			eDVBServicePMTHandler.eventNewProgramInfo: None,
 			eDVBServicePMTHandler.eventTuned: None,
