--- EVO/usr/lib/enigma2/python/Components/Language.py	2009-09-17 22:05:52.000000000 +0200
+++ EVO/usr/lib/enigma2/python/Components/Language.py	2009-09-28 12:38:22.947359834 +0200
@@ -13,32 +13,10 @@
 		# FIXME make list dynamically
 		# name, iso-639 language, iso-3166 country. Please don't mix language&country!
 		# also, see "precalcLanguageList" below on how to re-create the language cache after you added a language
-		self.addLanguage(_("English"), "en", "EN")
 		self.addLanguage(_("German"), "de", "DE")
-		self.addLanguage(_("Arabic"), "ar", "AE")
-		self.addLanguage(_("Catalan"), "ca", "AD")
-		self.addLanguage(_("Croatian"), "hr", "HR")
-		self.addLanguage(_("Czech"), "cs", "CZ")
-		self.addLanguage(_("Danish"), "da", "DK")
-		self.addLanguage(_("Dutch"), "nl", "NL")
-		self.addLanguage(_("Finnish"), "fi", "FI")
-		self.addLanguage(_("Estonian"), "et", "EE")
-		self.addLanguage(_("French"), "fr", "FR")
-		self.addLanguage(_("Greek"), "el", "GR")
-		self.addLanguage(_("Hungarian"), "hu", "HU")
-		self.addLanguage(_("Lithuanian"), "lt", "LT")
-		self.addLanguage(_("Latvian"), "lv", "LV")
-		self.addLanguage(_("Icelandic"), "is", "IS")
-		self.addLanguage(_("Italian"), "it", "IT")
+		self.addLanguage(_("English"), "en", "EN")
 		self.addLanguage(_("Norwegian"), "no", "NO")
-		self.addLanguage(_("Polish"), "pl", "PL")
-		self.addLanguage(_("Portuguese"), "pt", "PT")
 		self.addLanguage(_("Russian"), "ru", "RU")
-		self.addLanguage(_("Spanish"), "es", "ES")
-		self.addLanguage(_("Swedish"), "sv", "SE")
-		self.addLanguage(_("Turkish"), "tr", "TR")
-		self.addLanguage(_("Ukrainian"), "uk", "UA")
-		self.addLanguage(_("Frisian"), "fy", "x-FY") # there is no separate country for frisian
 
 		self.callbacks = []
 
